### DISCUSSION TITLE
useSWRInfinite with URL query change (such as search term)

### DISCUSSION THREAD

**Discussion by asaf** [vercel/swr#1107] [Q&A]
Hey,

When using `useSWRInfinite`, the `setSize` function preserves data, which is great as it avoids the page to jump until the new data is returned.

In our case, the query also expects a _search term_ that may be different between calls. but when the query parameter changes, the data gets wiped (becomes _undefined_ until data is validated) so the page jumps (data is cleaned) until data is back and then shown.

Is there a way to use `useSWRInfinite` in a way that not only the page size change while still preserves the same behavior?

Thanks,

Asaf.

**Comment by asaf** [comment]
please close, see [issue 192](https://github.com/vercel/swr/issues/192)

**Reply by shuding** [reply]
For `useSWRInfinite`, there's an option to persist the page size: https://swr.vercel.app/docs/pagination#parameters.
